File: springapp/src/main/java/com/examly/springapp/service/ShareService.java
package com.examly.springapp.service;

import com.examly.springapp.dto.ShareRequest;
import com.examly.springapp.dto.ShareResponse;
import com.examly.springapp.model.ShareLink;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface ShareService {
    ShareResponse createShareLink(ShareRequest request, Long createdBy);
    ShareResponse getShareByToken(String token);
    ResponseEntity<byte[]> downloadSharedDocument(String token);
    List<ShareResponse> getSharesByDocument(Long documentId, Long requesterId);
    List<ShareResponse> getSharesByUser(Long userId);
    void revokeShareLink(Long shareId, Long requesterId);
    void revokeByToken(String token, Long requesterId);
}

File: reactapp/src/services/shareService.js
import apiClient from './apiClient';

export const createShareLink = (requesterId, shareData) =>
  apiClient.post('/api/shares/create', shareData, { params: { requesterId } });

export const getShareByToken = (token) =>
  apiClient.get(`/api/shares/token/${token}`);

export const downloadSharedDocument = (token) =>
  apiClient.get(`/api/shares/download/${token}`, { responseType: 'blob' });

export const getSharesByDocument = (documentId, requesterId) =>
  apiClient.get(`/api/shares/document/${documentId}`, { params: { requesterId } });

export const getSharesByUser = (userId) =>
  apiClient.get(`/api/shares/user/${userId}`);

export const revokeShareLink = (shareId, requesterId) =>
  apiClient.delete(`/api/shares/${shareId}`, { params: { requesterId } });
